pss/crypto.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+package pss
+
+import (
+	"context"
+	"crypto/ecdsa"
+
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv6"
+)
+
+type CryptoBackend interface {
+	GetSymKey(id string) ([]byte, error)
+	GenerateSymKey() (string, error)
+	AddSymKeyDirect(bytes []byte) (string, error)
+	GetPrivateKey(id string) (*ecdsa.PrivateKey, error)
```

Comment:
I know this isn't a candidate for a final generic crypto API for pss, but whatever we do I doubt we will have a method that does this.

Reply:
I agree, and in fact it is only used in tests, but in all tests. What we can do is create a CryptoUtils interface for this kind of things.

Reply:
`GetPrivateKey`, `NewKeyPair` and `GenerateKey` moved to `CryptoUtils` that will be used only by tests.
In commit 49a3b06